include/net/net_if.h

```diff
@@ -442,6 +442,11 @@ struct net_if_dev {
 	struct net_offload *offload;
 #endif /* CONFIG_NET_OFFLOAD */
 
+#if defined(CONFIG_NET_SOCKETS_OFFLOAD)
```

Comment:
move it at the end, after mtu attribute

Reply:
Hmm, I didn't think of flag, this could save a byte. On the other hand, it sounds sane to keep this option separate, as it's conditionally enabled (and pretty niche to be honest).

Applied @tbursztyka for now, let me know if I should make a flag out of it still.

Reply:
> Hmm, I didn't think of flag, this could save a byte. On the other hand, it sounds sane to keep this option separate, as it's conditionally enabled (and pretty niche to be honest).

After a quick look, we are not saving anything in this case as the struct is not packed and there is still place for two extra bytes at the end. So we can go with the proposed route in this case.